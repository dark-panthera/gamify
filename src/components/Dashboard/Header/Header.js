import React from "react";
import { Link } from "react-router-dom";

const header = props => {
  return (
    <header className="header">
      <div className="">
        <div className="header__content">
          <div className="header__left">
            <div className="header__bar" onClick={props.onHideSidebar}>
              <div className="bar"></div>
              <div className="bar"></div>
              <div className="bar"></div>
            </div>
            <Link to="/dashboard" className="header__title">
              <h1>Netlify</h1>
            </Link>
          </div>

          <div className="header__right">
            <button className="button button--link" onClick={props.onToggleFilter}>
              <span className="fa fa-filter"></span>
            </button>
          </div>
        </div>
      </div>
    </header>
  );
};

export default header;
