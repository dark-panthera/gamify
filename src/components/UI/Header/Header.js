import React, { Component } from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";

class Header extends Component {
  render() {
    return (
      <div className="header">
        <div className="">
          <div className="header__content">
            <div className="header__left">
              <div className="header__bar" onClick={this.props.onHideSidebar}>
                <div className="bar"></div>
                <div className="bar"></div>
                <div className="bar"></div>
              </div>
              <Link to="/" className="header__title">
                <h1>Gamify</h1>
              </Link>
            </div>

            <div className="header__right">
              {this.props.isAuthenticated ? (
                <Link to="/logout" className="button button--link">
                  Logout
                </Link>
              ) : (
                <Link to="/login" className="button button--link">
                  Login
                </Link>
              )}
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  isAuthenticated: !!state.auth.uid
});

export default connect(mapStateToProps)(Header);
