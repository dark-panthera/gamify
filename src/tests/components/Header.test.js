import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/Dashboard/Header/Header';

test('should render Header correctly', () => {
  const wrapper = shallow(<Header onHideSidebar={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});

test("should call onHideSidebar on button click", () => {
  const onToggleFilter = jest.fn();
  const wrapper = shallow(<Header onToggleFilter={onToggleFilter} />);
  wrapper.find('button').simulate('click');
  expect(onToggleFilter).toHaveBeenCalled();
});
