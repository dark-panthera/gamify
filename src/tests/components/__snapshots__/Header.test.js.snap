// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header
  className="header"
>
  <div
    className=""
  >
    <div
      className="header__content"
    >
      <div
        className="header__left"
      >
        <div
          className="header__bar"
          onClick={[Function]}
        >
          <div
            className="bar"
          />
          <div
            className="bar"
          />
          <div
            className="bar"
          />
        </div>
        <Link
          className="header__title"
          to="/dashboard"
        >
          <h1>
            Netlify
          </h1>
        </Link>
      </div>
      <div
        className="header__right"
      >
        <button
          className="button button--link"
        >
          <span
            className="fa fa-filter"
          />
        </button>
      </div>
    </div>
  </div>
</header>
`;
